### ISSUE TITLE
Template selector in the Document Designer does not show existing templates for the current profile

### ISSUE THREAD

**Issue by pgathogo** [gltn/stdm#448]
In the document designer, the Template Selector dialog does not show existing templates for the current profile. When you open the selector it is empty.

**Comment by gkahiu** [comment]
Hi Paul,

I cannot reproduce, I can see the template that I saved for the current profile. See below:

![image](https://user-images.githubusercontent.com/3840267/111495778-c0fef580-8750-11eb-975c-cd5c19b6aace.png)

Perhaps it could also be a 'relatively' old issue addressed in the latest commits, you can try pull the latest master and confirm.

**Comment by pgathogo** [comment]
@gkahiu new templates are showing, however, for old templates for the same profile are not showing.

**Comment by gkahiu** [comment]
Yes, now I do confirm. There is now a huge shift in the internal mechanics of how custom attributes (e.g. data source, referenced table name etc.) are stored in a layout hence the reason as to why old (composer-based) templates are not being loaded.

@nyalldawson, the current logic is not able able to read the data source defined in older composer-based templates (see an example [here](https://github.com/gltn/stdm/blob/master/stdm/templates/Certificate%20of%20Garden%20Occupancy%20Sample.sdt)). I see there is a `QgsCompositionConverter` class for converting 2.x composition to 3.x layouts but its not available in Python bindings. Also am not sure how to do the conversions for STDM composer items (e.g. line, labels, attribute tables etc.) to the new custom layout items. Perhaps we might need something like an `STDMTemplateConverter` for updating STDM composer-based templates, please assist.
